tests/io/test_qualtran_io.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :mod:`pennylane.io.qualtran_io` module.
+"""
+import numpy as np
+import pennylane as qml
+
+
+class TestFromBloq:
+    """Test that FromBloq accurately wraps around Bloqs."""
+
+    def test_repr(self):
+        """Tests that FromBloq has the correct __repr__"""
+
+        from qualtran.bloqs.basic_gates import XGate
+
+        assert qml.FromBloq(XGate(), 1).__repr__() == "FromBloq(XGate, wires=Wires([1]))"
+
+    def test_composite_bloq_advanced(self):
+        """Tests that a composite bloq with higher level abstract bloqs has the correct
+        decomposition after wrapped with `FromBloq`"""
+        from qualtran import BloqBuilder
+        from qualtran import QUInt
+        from qualtran.bloqs.arithmetic import Product, Add
+        from pennylane.wires import Wires
+
+        bb = BloqBuilder()
+
+        w1 = bb.add_register("p1", 3)
+        w2 = bb.add_register("p2", 3)
+        w3 = bb.add_register("q1", 3)
+        w4 = bb.add_register("q2", 3)
+
+        w1, w2, res1 = bb.add(Product(3, 3), a=w1, b=w2)
+        w3, w4, res2 = bb.add(Product(3, 3), a=w3, b=w4)
+        p1p2, p1p2_plus_q1q2 = bb.add(Add(QUInt(bitsize=6), QUInt(bitsize=6)), a=res1, b=res2)
+
+        cbloq = bb.finalize(p1=w1, p2=w2, q1=w3, q2=w4, p1p2=p1p2, p1p2_plus_q1q2=p1p2_plus_q1q2)
+
+        expected = [
+            qml.FromBloq(Product(3, 3), wires=Wires([0, 1, 2, 3, 4, 5, 12, 13, 14, 15, 16, 17])),
+            qml.FromBloq(Product(3, 3), wires=Wires([6, 7, 8, 9, 10, 11, 18, 19, 20, 21, 22, 23])),
+            qml.FromBloq(
+                Add(QUInt(bitsize=6), QUInt(bitsize=6)),
+                wires=Wires([12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23]),
+            ),
+        ]
+        assert qml.FromBloq(cbloq, wires=range(24)).decomposition() == expected
+
+    def test_composite_bloq(self):
```

Comment:
consider adding a test that compares the unitary matrices
 - derived using `Bloq.tensor_contract()`
 - derived using `FromBloq(b).matrix()`
 - reference

using a non-atomic bloq b